zerver/lib/subscription_info.py

```diff
@@ -654,8 +678,10 @@ def gather_subscriptions_helper(
     for stream_dict in all_stream_dicts:
         for setting_name in Stream.stream_permission_group_settings:
             setting_group_ids.add(stream_dict[setting_name + "_id"])
-
-    setting_groups_dict = get_setting_values_for_group_settings(list(setting_group_ids))
+    anonymous_group_ids = UserGroup.objects.filter(
+        id__in=setting_group_ids, named_user_group=None
+    ).values_list("id", flat=True)
+    anonymous_group_membership = get_members_and_subgroups_of_groups(set(anonymous_group_ids))
```

Comment:
This is only used in one place, but may be we can extract this to a function.

Reply:
Yeah, giving that a name could be a useful refactor. We can wait until Shubham's PR is merged first.